packages/pymedphys_dicom/src/pymedphys_dicom/dicom/dose.py

```diff
@@ -32,100 +30,148 @@
+def depth_dose(depths, dose: pydicom.Dataset, plan: pydicom.Dataset):
+    depths = np.array(depths, copy=False)
 
-    depth = vertical + depth_adjust
+    surface_entry_point = get_surface_entry_point(plan)
+    depth_adjust = surface_entry_point.y
 
-    return inplane, crossplane, depth, dose
+    y = depths + depth_adjust
+    x, z = [surface_entry_point.x], [surface_entry_point.z]
 
+    coords = (z, y, x)
+
+    extracted_dose = np.squeeze(dicom_dose_interpolate(coords, dose))
+
+    return extracted_dose
+
+
+def profile(displacements, depth, direction, dose: pydicom.Dataset,
+            plan: pydicom.Dataset):
+
+    displacements = np.array(displacements, copy=False)
 
-def extract_depth_dose(ds, depth_adjust, averaging_distance=0):
-    inplane, crossplane, depth, dose = load_dicom_data(ds, depth_adjust)
+    surface_entry_point = get_surface_entry_point(plan)
+    depth_adjust = surface_entry_point.y
+    y = [depth + depth_adjust]
 
-    inplane_ref = abs(inplane) <= averaging_distance
-    crossplane_ref = abs(crossplane) <= averaging_distance
+    if direction is 'inplane':
+        coords = (
+            displacements + surface_entry_point.z,
+            y, [surface_entry_point.x]
+        )
+    elif direction is 'crossplane':
+        coords = (
+            [surface_entry_point.z], y,
+            displacements + surface_entry_point.x
+        )
+    else:
+        raise ValueError(
+            "Expected direction to be equal to 'inplane' or 'crossplane'")
+
+    extracted_dose = np.squeeze(dicom_dose_interpolate(coords, dose))
+
+    return extracted_dose
 
-    sheet_dose = dose[:, :, inplane_ref]
-    column_dose = sheet_dose[:, crossplane_ref, :]
 
-    depth_dose = np.mean(column_dose, axis=(1, 2))
+# def extract_depth_dose(ds, depth_adjust, averaging_distance=0):
+#     inplane, crossplane, depth, dose = load_dicom_data(ds, depth_adjust)
 
-    # uncertainty = np.std(column_dose, axis=(1, 2)) / depth_dose
-    # assert np.all(uncertainty < 0.01),
-    # "Shouldn't average over more than 1% uncertainty"
+#     inplane_ref = abs(inplane) <= averaging_distance
+#     crossplane_ref = abs(crossplane) <= averaging_distance
 
-    return depth, depth_dose
+#     sheet_dose = dose[:, :, inplane_ref]
+#     column_dose = sheet_dose[:, crossplane_ref, :]
 
+#     depth_dose = np.mean(column_dose, axis=(1, 2))
 
-def extract_profiles(ds, depth_adjust, depth_lookup, averaging_distance=0):
+#     # uncertainty = np.std(column_dose, axis=(1, 2)) / depth_dose
+#     # assert np.all(uncertainty < 0.01),
+#     # "Shouldn't average over more than 1% uncertainty"
 
-    inplane, crossplane, depth, dose = load_dicom_data(ds, depth_adjust)
+#     return depth, depth_dose
 
-    inplane_ref = abs(inplane) <= averaging_distance
-    crossplane_ref = abs(crossplane) <= averaging_distance
 
-    depth_reference = depth == depth_lookup
+# def extract_profiles(ds, depth_adjust, depth_lookup, averaging_distance=0):
```

Comment:
Possibly worth just deleting these functions instead of leaving them commented out.

Reply:
I think so :)

Reply:
Will do before the merge.